apps/managedidentity/managedidentity_test.go

```diff
@@ -4,43 +4,401 @@ package managedidentity
+		for _, value := range values {
+			queryString.Add(key, value)
+		}
 	}
+
+	return endpoint + "?" + queryString.Encode()
 }
 
-func TestManagedIdentityWithClaims(t *testing.T) {
-	client, err := fakeClient(ClientID("123"))
+func expectedRequest(source Source, resource string, id ID) (*http.Request, error) {
+	return expectedRequestWithId(source, resource, id)
+}
+
+func expectedRequestWithId(_ Source, resource string, id ID) (*http.Request, error) {
+	var endpoint string
+	headers := http.Header{}
+	queryParameters := make(map[string][]string)
+
+	//check with source when added different sources.
+	endpoint = imdsEndpoint
+	queryParameters["api-version"] = []string{"2018-02-01"}
+	queryParameters["resource"] = []string{resource}
+	headers.Add("Metadata", "true")
+
+	switch id.(type) {
+	case ClientID:
+		queryParameters[miQuerryParameterClientId] = []string{id.value()}
+	case ResourceID:
+		queryParameters[miQuerryParameterResourceId] = []string{id.value()}
+	case ObjectID:
+		queryParameters[miQuerryParameterObjectId] = []string{id.value()}
+	case systemAssignedValue:
+		// not adding anything
+	default:
+		return nil, fmt.Errorf("Type not supported")
+	}
 
+	uri, err := url.Parse(computeUri(endpoint, queryParameters))
 	if err != nil {
-		t.Fatal(err)
+		return nil, err
+	}
+
+	req := &http.Request{
+		Method: "GET",
+		URL:    uri,
+		Header: headers,
+	}
+
+	return req, nil
+}
+
+func ExpectedResponse(statusCode int, response string) http.Response {
+	return http.Response{
+		StatusCode: statusCode,
+		Body:       io.NopCloser(strings.NewReader(response)),
+	}
+}
+
+type resourceTestData struct {
+	source   Source
+	endpoint string
+	resource string
+}
+
+func createResourceData() []resourceTestData {
+	return []resourceTestData{
+		{source: DefaultToIMDS, endpoint: imdsEndpoint, resource: resource},
+		{source: DefaultToIMDS, endpoint: imdsEndpoint, resource: resourceDefaultSuffix},
+	}
+}
+
+func Test_SystemAssigned_Returns_Token_Failure(t *testing.T) {
+	testCases := []ErrorResponse{
+		{StatusCode: http.StatusNotFound, Message: ``, CorrelationID: "121212"},
+		{StatusCode: http.StatusNotImplemented, Message: ``, CorrelationID: "121212"},
+		{StatusCode: http.StatusServiceUnavailable, Message: ``, CorrelationID: "121212"},
+		{StatusCode: http.StatusBadRequest,
+			Message:       `{"error": "invalid_request", "error_description": "Identity not found"}`,
+			CorrelationID: "121212",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(strconv.Itoa(testCase.StatusCode), func(t *testing.T) {
+			fakeErrorClient := errorClient{errResponse: testCase}
+			client, err := fakeMIClient(SystemAssigned(), WithHTTPClient(&fakeErrorClient))
+			if err != nil {
+				t.Fatal(err)
+			}
+			resp, err := client.AcquireToken(context.Background(), resource)
+			if err == nil {
+				t.Fatalf("testManagedIdentity: Should have encountered the error")
+			}
+			switch e := err.(type) {
+			case errors.CallErr:
```

Comment:
Looks like these tests should fail when the error is a different type. However, as written they just skip error validation in that case. Try
```go
var e errors.CallErr
if errors.As(err, &e) {
	// TODO
} else {
	t.Fatalf("wanted %T, got %T", e, err)
}
```